src/Features/LanguageServer/Protocol/Handler/Diagnostics/DiagnosticSourceProviders/IDiagnosticSourceProvider.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Diagnostics;
+
+/// <summary>
+/// Provides diagnostic sources.
+/// </summary>
+internal interface IDiagnosticSourceProvider
+{
+    /// <summary>
+    /// True if this provider is for documents, false if it is for workspace.
+    /// </summary>
+    bool IsDocument { get; }
+
+    /// <summary>
+    /// Provider's name. Each should have a unique name within <see cref="IsDocument"/> scope.
+    /// </summary>
+    string Name { get; }
+
+    /// <summary>
+    /// Creates the diagnostic sources.
+    /// </summary>
+    /// <param name="context">The context.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
```

Comment:
```suggestion
```

Remove param docs that don't actually explain anything.  If they just say 'The X' they can be removed.